railgun/test/integration/ingress_test.go

```diff
@@ -309,3 +309,99 @@ func TestIngressClassNameSpec(t *testing.T) {
+	_, err = cluster.Client().CoreV1().Services(elsewhere).Create(ctx, service, metav1.CreateOptions{})
+	assert.NoError(t, err)
+	_, err = cluster.Client().CoreV1().Services(nowhere).Create(ctx, service, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up the service %s", service.Name)
+		assert.NoError(t, cluster.Client().CoreV1().Services(elsewhere).Delete(ctx, service.Name, metav1.DeleteOptions{}))
+		assert.NoError(t, cluster.Client().CoreV1().Services(nowhere).Delete(ctx, service.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("creating an ingress for service %s with ingress.class %s", service.Name, ingressClass)
+	elsewhereIngress := k8sgen.NewIngressForService("/elsewhere", map[string]string{
+		annotations.IngressClassKey: ingressClass,
+		"konghq.com/strip-path":     "true",
+	}, service)
+	nowhereIngress := k8sgen.NewIngressForService("/nowhere", map[string]string{
+		annotations.IngressClassKey: ingressClass,
+		"konghq.com/strip-path":     "true",
+	}, service)
+	elsewhereIngress, err = cluster.Client().NetworkingV1().Ingresses(elsewhere).Create(ctx, elsewhereIngress, metav1.CreateOptions{})
+	assert.NoError(t, err)
+	nowhereIngress, err = cluster.Client().NetworkingV1().Ingresses(nowhere).Create(ctx, nowhereIngress, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("ensuring that Ingress %s is cleaned up", elsewhereIngress.Name)
+		if err := cluster.Client().NetworkingV1().Ingresses(elsewhere).Delete(ctx, elsewhereIngress.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+		if err := cluster.Client().NetworkingV1().Ingresses(nowhere).Delete(ctx, nowhereIngress.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Logf("waiting for routes from Ingress %s to be operational", elsewhereIngress.Name)
+	p := proxyReady()
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/elsewhere", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			// now that the ingress backend is routable, make sure the contents we're getting back are what we expect
+			// Expected: "<title>httpbin.org</title>"
+			b := new(bytes.Buffer)
+			b.ReadFrom(resp.Body)
+			return strings.Contains(b.String(), "<title>httpbin.org</title>")
+		}
+		return false
+	}, ingressWait, waitTick)
+
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/nowhere", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		return expect404WithNoRoute(t, p.ProxyURL.String(), resp)
+	}, ingressWait, waitTick)
+}
```

Comment:
This is a good proof that you can enable/disable namespaces for support by the ctrl mgr, however do you think perhaps we should add one more namespace into the fray so we're proving the "multiple selective namespaces support"?

Reply:
This test is working with the assumption that other tests in the group have already tested config in the `default` namespace. Between it and the existing TestMinimalIngress we confirm that we can watch multiple. Not atomic, but it cuts down on a bunch of repetition in TestIngressNamespaces.